File: shared/src/commonMain/kotlin/com/slowthemdown/shared/model/VehicleReferences.kt
package com.slowthemdown.shared.model

data class VehicleReference(
    val name: String,
    val lengthMeters: Double,
    val category: VehicleCategory
)

object VehicleReferences {
    val all: List<VehicleReference> = listOf(
        // Sedans
        VehicleReference("Toyota Camry", 4.877, VehicleCategory.SEDAN),
        VehicleReference("Honda Civic", 4.694, VehicleCategory.SEDAN),
        VehicleReference("Honda Accord", 4.938, VehicleCategory.SEDAN),
        VehicleReference("Toyota Corolla", 4.663, VehicleCategory.SEDAN),
        VehicleReference("Nissan Altima", 4.846, VehicleCategory.SEDAN),
        VehicleReference("Hyundai Sonata", 4.846, VehicleCategory.SEDAN),
        // SUVs
        VehicleReference("Toyota RAV4", 4.602, VehicleCategory.SUV),
        VehicleReference("Honda CR-V", 4.602, VehicleCategory.SUV),
        VehicleReference("Ford Explorer", 5.090, VehicleCategory.SUV),
        VehicleReference("Chevy Equinox", 4.663, VehicleCategory.SUV),
        VehicleReference("Toyota Highlander", 4.938, VehicleCategory.SUV),
        VehicleReference("Jeep Grand Cherokee", 4.907, VehicleCategory.SUV),
        // Trucks
        VehicleReference("Ford F-150 (Crew Cab)", 5.883, VehicleCategory.TRUCK),
        VehicleReference("Chevy Silverado (Crew)", 6.035, VehicleCategory.TRUCK),
        VehicleReference("RAM 1500 (Crew Cab)", 5.822, VehicleCategory.TRUCK),
        VehicleReference("Toyota Tacoma (Double)", 5.425, VehicleCategory.TRUCK),
        VehicleReference("Ford Ranger (Crew)", 5.334, VehicleCategory.TRUCK),
        // Vans
        VehicleReference("Honda Odyssey", 5.151, VehicleCategory.VAN),
        VehicleReference("Toyota Sienna", 5.151, VehicleCategory.VAN),
        VehicleReference("Chrysler Pacifica", 5.182, VehicleCategory.VAN),
        // Compact
        VehicleReference("VW Golf", 4.297, VehicleCategory.COMPACT),
        VehicleReference("Mazda 3", 4.572, VehicleCategory.COMPACT),
        VehicleReference("Subaru Impreza", 4.511, VehicleCategory.COMPACT),
    )

    fun byCategory(): List<Pair<VehicleCategory, List<VehicleReference>>> =
        VehicleCategory.entries.mapNotNull { cat ->
            val vehicles = all.filter { it.category == cat }
            if (vehicles.isEmpty()) null else cat to vehicles
        }

    /** Average vehicle length in meters */
    const val averageLength: Double = 4.724
}

File: ios/SlowThemDown/Models/VehicleReferences.swift
import Foundation

struct VehicleReference: Identifiable, Hashable {
    let id = UUID()
    let name: String
    let lengthMeters: Double
    let category: VehicleCategory
}

enum VehicleCategory: String, CaseIterable {
    case sedan = "Sedan"
    case suv = "SUV"
    case truck = "Truck"
    case van = "Van"
    case compact = "Compact"
}

enum VehicleReferences {
    static let all: [VehicleReference] = [
        // Sedans
        VehicleReference(name: "Toyota Camry", lengthMeters: 4.877, category: .sedan),
        VehicleReference(name: "Honda Civic", lengthMeters: 4.694, category: .sedan),
        VehicleReference(name: "Honda Accord", lengthMeters: 4.938, category: .sedan),
        VehicleReference(name: "Toyota Corolla", lengthMeters: 4.663, category: .sedan),
        VehicleReference(name: "Nissan Altima", lengthMeters: 4.846, category: .sedan),
        VehicleReference(name: "Hyundai Sonata", lengthMeters: 4.846, category: .sedan),

        // SUVs
        VehicleReference(name: "Toyota RAV4", lengthMeters: 4.602, category: .suv),
        VehicleReference(name: "Honda CR-V", lengthMeters: 4.602, category: .suv),
        VehicleReference(name: "Ford Explorer", lengthMeters: 5.090, category: .suv),
        VehicleReference(name: "Chevy Equinox", lengthMeters: 4.663, category: .suv),
        VehicleReference(name: "Toyota Highlander", lengthMeters: 4.938, category: .suv),
        VehicleReference(name: "Jeep Grand Cherokee", lengthMeters: 4.907, category: .suv),

        // Trucks
        VehicleReference(name: "Ford F-150 (Crew Cab)", lengthMeters: 5.883, category: .truck),
        VehicleReference(name: "Chevy Silverado (Crew)", lengthMeters: 6.035, category: .truck),
        VehicleReference(name: "RAM 1500 (Crew Cab)", lengthMeters: 5.822, category: .truck),
        VehicleReference(name: "Toyota Tacoma (Double)", lengthMeters: 5.425, category: .truck),
        VehicleReference(name: "Ford Ranger (Crew)", lengthMeters: 5.334, category: .truck),

        // Vans
        VehicleReference(name: "Honda Odyssey", lengthMeters: 5.151, category: .van),
        VehicleReference(name: "Toyota Sienna", lengthMeters: 5.151, category: .van),
        VehicleReference(name: "Chrysler Pacifica", lengthMeters: 5.182, category: .van),

        // Compact
        VehicleReference(name: "VW Golf", lengthMeters: 4.297, category: .compact),
        VehicleReference(name: "Mazda 3", lengthMeters: 4.572, category: .compact),
        VehicleReference(name: "Subaru Impreza", lengthMeters: 4.511, category: .compact),
    ]

    static func byCategory() -> [(category: VehicleCategory, vehicles: [VehicleReference])] {
        VehicleCategory.allCases.compactMap { cat in
            let vehicles = all.filter { $0.category == cat }
            return vehicles.isEmpty ? nil : (cat, vehicles)
        }
    }

    /// Average vehicle length in meters
    static let averageLength: Double = 4.724
}
